zabbixsender/zabbixsvc/zabbixsvc.go

```diff
@@ -101,11 +135,26 @@ func (h *JSONHandler) zabbixSend(metrics []*zabbixsnd.Metric) (*ZabbixResponse,
 	if err := json.Unmarshal(res[13:], &zres); err != nil {
 		return nil, err
 	}
-
 	infoSplit := strings.Split(zres.Info, " ")
-	if strings.Trim(infoSplit[3], ";") != "0" || zres.Response != "success" {
-		return nil, errors.New("failed to fulfill the requests: " + strings.Trim(infoSplit[5], ";"))
+
+	failed := strings.Trim(infoSplit[3], ";")
+	if failed != "0" || zres.Response != "success" {
+		return nil, errors.Errorf("failed to fulfill the requests: %v", failed)
 	}
 
 	return &zres, nil
 }
+
+func (h *JSONHandler) LoadHostsFromFile(filename string) error {
+	hostsFile, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return errors.Wrapf(err, "can't open the alerts file- %v", filename)
+	}
+
+	err = yaml.Unmarshal(hostsFile, &h.Hosts)
```

Comment:
Wasn't sure about this part, is it better to unmarshal straight to the struct JSONHandler member, or to variable hosts map[string]string and then return the hosts from func and only then assign it to the JSONHandler. As you often said that function(for example Send) shoud only do the main functionally it was meant for.